Carbon.IIS/Functions/Set-CIisCollection.ps1

```diff
@@ -0,0 +1,119 @@
+
+        <location path="SITE_NAME">
+            <system.webServer>
+                <defaultDocument>
+                    <files>
+                        <clear />
+                        <add value="default.aspx" />
+                        <add value="index.html" />
+                    </files>
+                </defaultDocument>
+            </system.webServer>
+        </location>
+
+    .EXAMPLE
+    @{ name = 'HttpLoggingModule' ; image = '%windir%\System32\inetsrv\loghttp.dll' } | Set-CIisCollection -SectionPath 'system.webServer/globalModules'
+
+    Demonstrates how to set a collection to a specific list where each item in the collection has more than a single
+    default attribute. The above command would result in the applicationHost.config file to be updated to be:
+
+        <system.webServer>
+            <globalModules>
+                ...
+                <add name="HttpLoggingModule" image="%windir%\System32\inetsrv\loghttp.dll" />
+            </globalModules>
+        </system.webServer>
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory, ParameterSetName='Location')]
+        [String] $LocationPath,
+
+        [Parameter(Mandatory)]
+        [String] $SectionPath,
+
+        # If no name, call `GetCollection()` on the configuration element, passing no name.
+        [String] $Name,
+
+        # Pass a hashtable for each item that has more than one attribute value to set. Otherwise, pass the attribute
+        # value for the default attribute.
+        [Parameter(ValueFromPipeline)]
+        [Object[]] $InputObject
+    )
+
+    begin
+    {
+        Set-StrictMode -Version 'Latest'
+        Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+        $items = [Collections.Generic.List[Microsoft.Web.Administration.ConfigurationElement]]::New()
+        $collection = Get-CIisCollection @PSBoundParameters
+        $keyAttrName = Get-CIisCollectionKeyName -Collection $collection
+        Write-Verbose $keyAttrName
+        $save = $false
+    }
+
+    process
+    {
+        Write-Verbose 'process'
+        $addItem = $collection.CreateElement('add')
+        if ($InputObject[0] -is [string])
+        {
+            Write-Verbose 'is a string'
+            $addItem.SetAttributeValue($keyAttrName, $InputObject[0])
+        }
+        elseif ($InputObject[0] -is [hashtable])
+        {
+            Write-Verbose 'is a hashtable'
+            foreach ($key in $InputObject[0].Keys)
+            {
+                $addItem.SetAttributeValue($key, $InputObject[0][$key])
+            }
+        }
+        else {
+            Write-Verbose 'is neither'
+            $InputObject.GetType() | Write-Verbose
+        }
+        $items.Add($addItem)
+        Write-Verbose $items.Count
+    }
+    end
+    {
+        foreach ($collectionItem in $collection)
+        {
+            if ($collectionItem -notin $items -and $LocationPath)

```

Comment:
Why the `-and $LocationPath` statement? That shouldn't matter?

Reply:
I was under the impression that I shouldn't modify anything in the default set so I was only clearing the collection *if* it was for a site. I should change where the `$save` variable is set though. It should be set as long as there are things being modified.

Reply:
The purpose of `Set-CIisCollection` is to ensure a collection contains *only* the items passed in, regardless of site or global.